wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -137,6 +142,23 @@ instance
     getNewAddress (accId, passphrase) = do
         clientAddress <- L.newAddress RandomSeed passphrase accId
         decodeCTypeOrFail (cadId clientAddress)
+    -- We are creating a sample HD address here. It has the same
+    -- length (76) as the biggest one created by Daedalus.
+    getFakeChangeAddress = do
```

Comment:
I don't like this function is tail recursive. You suggest believing it will work and find an address quickly instead of using a predefined constant address with deterministic guarantees. There are already a lot of kostils in the wallet which we just believe.

I insist on rewriting it to deterministic manner.